components/ee/payment-endpoint/src/accounting/subscription-service.ts

```diff
@@ -85,6 +85,27 @@ export class SubscriptionService {
         });
     }
 
+    /**
+     * Subscribes the given user to the "Professional Open Source" plan if they don't already have either:
+     *  - the Professional Open Source plan
+     *  - another paid plan
+     * @param user
+     * @param now
+     * @returns a subscription if that is blocking assignment of the OSS subscription
+     */
+    async checkAndSubscribeToOssSubscription(user: User, now: Date): Promise<Subscription | undefined> {
+        const userId = user.id;
+
+        const subs = await this.getNotYetCancelledSubscriptions(user, now.toISOString());
+        const uncancelledPaidOrOssSub = subs.find(s => (!Plans.isFreePlan(s.planId) || s.planId === Plans.FREE_OPEN_SOURCE.chargebeeId) && !s.cancellationDate);
+        if (uncancelledPaidOrOssSub) {
+            return uncancelledPaidOrOssSub;
+        }
+
+        await this.subscribe(userId, Plans.FREE_OPEN_SOURCE, undefined, now.toISOString());
+        return undefined;
```

Comment:
I think there is no need to check for other (paid) subscriptions. It's actually fine to have both (Professional Open Source + a paid plan), see for example my plans:

<img width="1438" alt="Screenshot 2021-12-14 at 08 52 48" src="https://user-images.githubusercontent.com/599268/145955940-85c0138b-b68f-4f74-af56-fe10e76a5451.png">

Having both allows to benefit from the Professional Open Source "status" even if you already pay for another plan.

Reply:
@jankeromnes Do we display that independently? That's news to me! And the screenshot above shows a "Team Subscription". Do we handle/display OSS the same, really? 
I wanted to avoid situations where we "override" peoples paid subscription, and they end up not noticing because of this.

Reply:
Yes, we display them independently.

- Professional Open Source is a sort of "status" (if you have it, it replaces your "free plan" card on the left -- as in my screenshot)
- The screenshot above shows a "Team Subscription", because I'm part of a "Team Unleashed" plan -- but that's actually not relevant (if I were on the regular "Unleashed" plan, it would show that as my active plan, without a mention of a team plan, and I'd still have "Professional Open Source" as my left-most card)

Reply:
Removed the paid-plan check.

I kept the check for "Prof. OSS", though, to not cancel/add a new subscription on each login.